clang/test/CodeGen/builtins-nvptx.c

```diff
@@ -294,6 +294,81 @@ __device__ void nvvm_atom(float *fp, float f, double *dfp, double df, int *ip,
   // CHECK: call i32 @llvm.nvvm.atomic.load.dec.32.p0i32
   __nvvm_atom_dec_gen_ui(uip, ui);
 
+  // CHECK: call i32 @llvm.nvvm.ld.gen.i.volatile.i32.p0i32
+  __nvvm_volatile_ld_gen_i(ip);
```

Comment:
It would be great to attempt actually assembling PTX we generate for the tests.
In the past it helped a lot to catch cases where we were generating something that `ptxas` would not accept.

Reply:
This code is also tested in llvm-test-suite, where it is compiled, assembled and run. See https://github.com/intel/llvm-test-suite/pull/648

Reply:
If "assembling PTX" doesn't require NV HW, so we can put such tests to clang/test/*, it's better than running SYCL application.
Running SYCL application is more expensive as it requires NV HW, whereas clang/test/* allow developers to catch bugs much earlier in their development pipeline.

Reply:
> If "assembling PTX" doesn't require NV HW, so we can put such tests to clang/test/*

We'll still need `ptxas` so such test can't be part of LLVM/clang's own tests. Test-suite is the right place.